npc.cpp

```diff
@@ -89,15 +108,23 @@ void Game::GrowWMons(int rate)
 								 (ItemDefs[mon].flags & ItemType::DISABLED)))
 								avail = 1;
 
+							if (!avail) {
+								printf("\n\n NO AVAL  !!!!!!!!!!! \n\n");
+							}
 							if (avail)
 								wanted += TerrainDefs[reg->type].wmonfreq;
 						}
 					}
 				}
 
+				printf("\n\n GrowWMons 2: %d \n\n", wanted);
+
 				wanted /= 10;
 				wanted -= mons;
 				wanted = (wanted*rate + getrandom(100))/100;
+
+				printf("\n\n GrowWMons 3: %d . mons: %d. wanted: %d. rate: %d \n\n", wanted, mons, wanted, rate);
+
 				if (wanted > 0) {
 					for (int i=0; i< wanted;) {
 						int m=getrandom(8);
```

Comment:
This section of code that actually adds the monsters will also need to know about this change, or it could skew monster spawn location distributions (if a level is 5 hexes across using 8 as the parameter to random will make the first 3 columns twice as likely to receive wandering monsters as the next 5, because of wrapping)

Reply:
Also: I don't like this "pick random locations until you find a valid one" algorithm because it has very bad worst-case behaviour (potentially infinite if you have a dodgy RNG), which is why the code usually does the enumerate-possibilities-and-choose-once method.  I know you didn't write this code, but it might be worth fixing while we're here?

Reply:
sure, i will try to fix this as well